src/common/__tests__/kube-helpers.test.ts

```diff
@@ -0,0 +1,180 @@
+      expect(logger.error).not.toBeCalled();
+      expect(kc.getCurrentContext()).toBe('minikube');
+    });
+
+    it("multiple context is ok", async () => {
+      mockKubeConfig.contexts.push({context: {cluster: "cluster-2", user: "cluster-2"}, name: "cluster-2"});
+      const kc:KubeConfig = loadConfig(JSON.stringify(mockKubeConfig));
+      expect(logger.error).not.toBeCalled();
+      expect(kc.getCurrentContext()).toBe('minikube');
+      expect(kc.contexts.length).toBe(2);
+    });
+  });
+
+  describe('Check invalid kubeconfigs', () => {
+    beforeEach(() => {
+      mockKubeConfig = {
+        apiVersion: "v1",
+        clusters: [{
+          name: "minikube",
+          cluster: {
+            server: "https://192.168.64.3:8443",
+          },
+        }],
+        contexts: [{
+          context: {
+            cluster: "minikube",
+            user: "minikube",
+          },
+          name: "minikube",
+        }],
+        users: [{
+          name: "minikube",
+        }],
+        kind: "Config",
+        'current-context': "minikube",
+        preferences: {},
+      };
+    });
+
+    it("empty name in context causes it to be removed", async () => {
+      mockKubeConfig.contexts.push({context: {cluster: "cluster-2", user: "cluster-2"}, name: ""});
+      expect(mockKubeConfig.contexts.length).toBe(2);
+      const kc:KubeConfig = loadConfig(JSON.stringify(mockKubeConfig));
+      expect(logger.error).toBeCalledWith("loadFromString: context {\"context\":{\"cluster\":\"cluster-2\",\"user\":\"cluster-2\"},\"name\":\"\"} in kubeconfig has an error so will not be made available to Lens");
+      expect(kc.getCurrentContext()).toBe('minikube');
+      expect(kc.contexts.length).toBe(1);
+    });
+
+    it("empty cluster in context causes it to be removed", async () => {
+      mockKubeConfig.contexts.push({context: {cluster: "", user: "cluster-2"}, name: "cluster-2"});
+      expect(mockKubeConfig.contexts.length).toBe(2);
+      const kc:KubeConfig = loadConfig(JSON.stringify(mockKubeConfig));
+      expect(logger.error).toBeCalledWith("loadFromString: context {\"context\":{\"cluster\":\"\",\"user\":\"cluster-2\"},\"name\":\"cluster-2\"} in kubeconfig has an error so will not be made available to Lens");
+      expect(kc.getCurrentContext()).toBe('minikube');
+      expect(kc.contexts.length).toBe(1);
+    });
+
+    it("empty user in context causes it to be removed", async () => {
+      mockKubeConfig.contexts.push({context: {cluster: "cluster-2", user: ""}, name: "cluster-2"});
+      expect(mockKubeConfig.contexts.length).toBe(2);
+      const kc:KubeConfig = loadConfig(JSON.stringify(mockKubeConfig));
+      expect(logger.error).toBeCalledWith("loadFromString: context {\"context\":{\"cluster\":\"cluster-2\",\"user\":\"\"},\"name\":\"cluster-2\"} in kubeconfig has an error so will not be made available to Lens");
+      expect(kc.getCurrentContext()).toBe('minikube');
+      expect(kc.contexts.length).toBe(1);
+    });
```

Comment:
should also test missing name/cluster/user? I tried a config file with typos:
```
contexts:
- context:
    scluster: docker-desktop
    user: docker-desktop
  name: docker-desktop
- context:
    cluster: kind-jim
    suser: kind-jim
  name: kind-jim
- context:
    cluster: minikube
    user: minikube
  sname: minikube
```
I was surprised to find that the `kind-jim` cluster was accepted as valid. There must be a default user(?)